src/callbacks_step/amr_dg2d.jl

```diff
@@ -112,10 +129,34 @@ function refine!(u_ode::AbstractVector, adaptor, mesh::Union{TreeMesh{2}, P4estM
                 # Refine element and store solution directly in new data structure
                 refine_element!(u, element_id, old_u, old_element_id,
                                 adaptor, equations, dg)
+
+                if mesh isa P4estMesh
+                    # Before `element_id` is incremented, divide by the new Jacobians on each
+                    # child element and save the result
+                    for m in 0:3 # loop over the children
+                        for v in eachvariable(equations)
+                            u[v, .., element_id + m] .*= (0.25 .*
```

Comment:
```suggestion
                            u[v, .., element_id + m] .*= (0.25f0 .*
```

To make it work with `Float32`, too

Reply:
Yes - if you can use integers, please do so 👍